ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1180 @@
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#renderFullPathButtons
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     */
+    public renderFullPathButtons(this: Breadcrumbs): void {
+        const breadcrumbs = this,
+            chart = breadcrumbs.chart,
+            breadcrumbsList = breadcrumbs.breadcrumbsList,
+            breadcrumbsOptions = breadcrumbs.options,
+            // last breadcrumb that was rendered.
+            lastBreadcrumbs = breadcrumbsList[breadcrumbsList.length - 2] &&
+                breadcrumbsList[breadcrumbsList.length - 2][2],
+            buttonSpacing = breadcrumbsOptions.buttonSpacing;
+
+        // Draw breadcrumbs.
+        if (breadcrumbs.breadcrumbsGroup && breadcrumbsList) {
+
+            // Make sure that only one type of button is visible.
+            if (chart.drillUpButton) {
+                chart.drillUpButton.destroy();
+                chart.drillUpButton = void 0;
+            }
+
+            // Inital position for calculating the breadcrumbsGroup.
+            let posX: number = lastBreadcrumbs ? lastBreadcrumbs.x +
+                    lastBreadcrumbs.element.getBBox().width + buttonSpacing : 0;
+            const posY: number = buttonSpacing;
+            breadcrumbsList.forEach(function (breadcrumb, index): void {
+                const breadcrumbButton: SVGElement = breadcrumb[2],
+                    separatorElement: SVGElement = breadcrumb[3];
+
+                // Render a button.
+                if (!breadcrumbButton) {
+                    const button = breadcrumbs.renderButton(breadcrumb, posX, posY);
+                    breadcrumb.push(button);
+                    posX += button ? button.getBBox().width + buttonSpacing : 0;
+                    breadcrumbs.breadcrumbsGroup.lastXPos = posX;
+                }
+
+                // Render a separator.
+                if (!separatorElement && index !== breadcrumbsList.length - 1) {
+                    const separator = breadcrumbs.renderSeparator(posX, posY);
+
+                    breadcrumb.push(separator);
+                    posX += separator ? separator.getBBox().width + buttonSpacing : 0;
+                }
+            });
+        }
+    }
+
+    /**
+     * Create a group, then draw breadcrumbs together with the separators.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#renderSingleButton
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     */
+    public renderSingleButton(this: Breadcrumbs): void {
+        const breadcrumbs = this,
+            chart = breadcrumbs.chart,
+            breadcrumbsList = breadcrumbs.breadcrumbsList,
+            breadcrumbsOptions = breadcrumbs.options,
+            lastBreadcrumbs = breadcrumbsList[breadcrumbsList.length - 2] &&
+                breadcrumbsList[breadcrumbsList.length - 2][2],
```

Comment:
Q: Why sometimes it refers to the last breadcrumb as `length - 2, and sometimes `length - 1`?

Reply:
Sometimes we want to refer to the last drawn breadcrumb (before drawing the new one) or we want to get previous breadcrumbs connector - then we use -2. 

It would be much more clear when switching naming to this.level - then it will be easier to understand from what levels we are taking parameters.